engine/src/main/java/org/hibernate/validator/cfg/context/PropertyTarget.java

```diff
@@ -30,6 +30,45 @@
 	 * @param type The access type (field/property).
 	 *
 	 * @return A creational context representing the selected property.
+	 *
+	 * @deprecated Since 6.1. Planned for removal. Use one of {@link PropertyTarget#field(String)},
+	 * {@link PropertyTarget#getter(String)} instead, based on a property kind.
 	 */
+	@Deprecated
 	PropertyConstraintMappingContext property(String property, ElementType type);
+
+	/**
+	 * Selects a filed to which the next operations shall apply.
+	 * <p>
+	 * Until this method is called constraints apply on class level. After calling this method constraints
+	 * apply on the specified field property.
+	 * </p>
+	 * <p>
+	 * A given field may only be configured once.
+	 * </p>
+	 *
+	 * @param property The field name, that represents a property on which to apply the following constraints.
+	 *
+	 * @return A creational context representing the selected field property.
+	 *
+	 */
+	PropertyConstraintMappingContext field(String property);
+
+	/**
+	 * Selects a getter to which the next operations shall apply.
+	 * <p>
+	 * Until this method is called constraints apply on class level. After calling this method constraints
+	 * apply on the specified getter property.
+	 * </p>
+	 * <p>
+	 * A given getter may only be configured once.
+	 * </p>
+	 *
+	 * @param property The getter property name (Java Bean notation), that represents a property on which
+	 * to apply the following constraints.
```

Comment:
I think we should give an example. Something like: "(using the JavaBean convention, e.g. {@code name} to address {@code getName()})".